packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -31,6 +33,16 @@ Future<Depfile> copyAssets(Environment environment, Directory outputDirectory) a
     pubspecFile,
   ];
   final List<File> outputs = <File>[];
+
+  final IconTreeShaker iconTreeShaker = IconTreeShaker(
```

Comment:
The targets that call this method need to add the iconTreeShaker deps to the inputs, specifically:

The app.dill if it isn't already there
The artifact kernel parser
The artifact font subseter

Reply:
This is not a target.

Reply:
I'll add these to the assets target

Reply:
There is also the targets that just call assets, unfortunately they're not all implemented with the same target (yet)

Reply:
So is just adding the sources to the assets targets ok or not?

Reply:
No, every target that calls that method needs the dependencies. For example: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/build_system/targets/android.dart#L22